docs/source/developers/cpp/development.rst

```diff
@@ -92,32 +92,51 @@ Address Sanitizer and Undefined Behavior Sanitizer to check for various
 patterns of misbehaviour such as memory leaks. In addition, the
 codebase is subjected to a number of code style and code cleanliness checks.
 
-In order to have a passing CI build, your modified git branch must pass the
+In order to have a passing CI build, your modified Git branch must pass the
 following checks:
 
 * C++ builds with the project's active version of ``clang`` without
   compiler warnings with ``-DBUILD_WARNING_LEVEL=CHECKIN``. Note that
   there are classes of warnings (such as ``-Wdocumentation``, see more
   on this below) that are not caught by ``gcc``.
+* Passes various C++ (and others) style checks, checked with the ``lint``
+  subcommand to :ref:`Archery <archery>`. This can also be fixed locally
+  by running ``archery lint --cpplint --clang-format --clang-tidy --fix``.
 * CMake files pass style checks, can be fixed by running
   ``archery lint --cmake-format --fix``. This requires Python
   3 and `cmake_format <https://github.com/cheshirekow/cmake_format>`_ (note:
-  this currently does not work on Windows)
-* Passes various C++ (and others) style checks, checked with the ``lint``
-  subcommand to :ref:`Archery <archery>`. This can also be fixed locally
-  by running ``archery lint --cpplint --fix``.
+  this currently does not work on Windows).
 
 In order to account for variations in the behavior of ``clang-format``
 between major versions of LLVM, we pin the version of ``clang-format``
-used. You can confirm the current pinned version by finding
-the ``CLANG_TOOLS`` variable value in `.env
-<https://github.com/apache/arrow/blob/master/.env>`_.
+used. You can confirm the current pinned version by finding the
+``CLANG_TOOLS`` variable value in `.env
+<https://github.com/apache/arrow/blob/master/.env>`_. Note that the version
+must match exactly; a newer version will not work. LLVM can be installed
+through a system package manager or a package manager like Conda or
+Homebrew, or binaries can be directly downloaded for various platforms from
+the `LLVM website <https://releases.llvm.org/>`_.
```

Comment:
@lidavidm I had to use the llvm12 and the homebrew only installed 12.0.1 (stable), as you suggested, we had to use the pre-built binaries. And it did work really well. Should we include that?

Reply:
That's what this is saying right?

Reply:
Yes, the idea is very clear, just more specific about the homebrew thing, it may not have the exact matching version as the default.